accounts/usbwallet/ledger.go

```diff
@@ -353,7 +367,9 @@ func (w *ledgerDriver) ledgerSign(derivationPath []uint32, tx *types.Transaction
 	// Chunk size selection to mitigate an underlying RLP deserialization issue on the ledger app.
 	// https://github.com/LedgerHQ/app-ethereum/issues/409
 	chunk := 255
-	for ; len(payload)%chunk <= ledgerEip155Size; chunk-- {
+	if tx.Type() == types.LegacyTxType {
```

Comment:
I am not sure if this is need for DynamicTx

Reply:
The workaround can be left in place for all tx types. It's just a way of modifying the transfer so it doesn't trigger a certain bug in the firmware.

Reply:
@fjl this doesn't seems to work for other tx.  This could very well be due to the way typed tx are formed. 
```
if tx.Type() == types.DynamicFeeTxType {
	if txrlp, err = rlp.EncodeToBytes([]interface{}{chainID, tx.Nonce(), tx.GasTipCap(), tx.GasFeeCap(), tx.Gas(), tx.To(), tx.Value(), tx.Data(), tx.AccessList()}); err != nil {
		return common.Address{}, nil, err
	}
	// append type to transaction
	txrlp = append([]byte{tx.Type()}, txrlp...)
}
```

I am not sure how to handle this bug here.
Ledger throws error
```
transaction type not supported
```

Reply:
So, I was going to ask you to show some snippet from ledger-docs or something, to demonstrate that they do accept this format, where you append the type into the payload like this. 

But, seeing as the ledger throws `transaction type not supported` -- doesn't this mean that this PR is moot, and that the ledger (at least with the software version you are using) does not support signing non-legacy transactions?

Reply:
@holiman So the error `transaction type not supported` appears when I apply `for ; len(payload)%chunk <= ledgerEip155Size; chunk--` logic for non legacy tx. If I don't do it - Ledger signing works.